registration/src/main/java/lobby/registration/OrderRepository.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package lobby.registration;
+
+import lobby.contracts.common.OrderId;
+import lobby.registration.service.OrderPricingService;
+import org.spine3.server.aggregate.AggregateRepositoryBase;
+
+import javax.annotation.Nonnull;
+
+/**
+ * The repository for order aggregate roots.
+ *
+ * @see OrderAggregate
+ * @see AggregateRepositoryBase
+ * @author Alexander Litus
+ */
+public class OrderRepository extends AggregateRepositoryBase<OrderId, OrderAggregate> {
+
+    private final OrderPricingService orderPricingService;
+
+    /**
+     * Creates a new repository instance.
+     *
+     * @param orderPricingService the pricing service to inject to order aggregate roots
+     */
+    public OrderRepository(OrderPricingService orderPricingService) {
+        super();
+        this.orderPricingService = orderPricingService;
+    }
+
+    @Nonnull
+    @Override
+    public OrderAggregate load(OrderId id) throws IllegalStateException {
```

Comment:
This method must be documented. Why do we override the default behaviour of `load()`?